CSLabs.Api/Controllers/ModuleController.cs

```diff
@@ -117,9 +134,34 @@ public async Task<IActionResult> Upsert([FromBody] Module module)
                 DatabaseContext.Update(module);
             else
                 DatabaseContext.Add(module);
+            // add tag relationships
+            module.AddModuleTags(DatabaseContext, module.ModuleTags);
             await DatabaseContext.SaveChangesAsync();
             await DatabaseContext.Entry(module).Collection(m => m.Labs).LoadAsync();
-            return Ok(module);
+            return Ok( 
+                await DatabaseContext.Modules
+                    .AsNoTracking()
+                    .Where(m => m.Id == module.Id)
+                    .IncludeTags()
+                    .ToListAsync()
+            );
+        }
+
+        [HttpDelete("tags")]
+        public async Task<IActionResult> DeleteTag(ModuleTag[] moduleTags)
```

Comment:
This method needs to be merged with the Upsert method instead, you can still have it a separate method, it just needs to be called internally instead from the api.